### PR TITLE
Recurring Payments: Open new window to connect instead of loading in place

### PR DESCRIPTION
**Pull request by apeatling**
…and connection flows in the same window.

Partially fixes https://github.com/Automattic/view-design/issues/275

#### Changes proposed in this Pull Request:

* When inside the widget editor or customizer view we do not want to open links within the same window. This will cause issues with losing design changes. This PR will check to see if we are in the customizer or widget editor and open a new window instead, where the flow can be completed.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Go to the customizer using the WordPress 5.8 branch on WordPress.com (use `svn switch` and ping me for the branch URL)
* On a sandboxed site go to the customizer and then to widgets > [widget area]
* Insert a payments block
* Click on the "Connect Stripe" option in the block toolbar.
* Confirm this opens a new window.
* Do the same in the post and site editors and confirm it still loads in the same window.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/shared/use-autosave-and-redirect/index.js [hunk 1]
```diff
@@ -31,6 +32,14 @@ export default function useAutosaveAndRedirect( redirectUrl, onRedirect = noop )
 
 	const isPostEditor = Object.keys( currentPost ).length > 0;
 
+	const isWidgetEditor = useSelect( select => {
```
Comment (stacimc) [inline projects/plugins/jetpack/extensions/shared/use-autosave-and-redirect/index.js:35]: Nice!

**Review by stacimc — COMMENTED** [review]
So far I have not been able to test this in the site-editor; happy to approve once I can get that working.

This looks good to me and tested well:

- [x] confirmed the bug on 5.8
- [x] confirmed that the Stripe connection opens in a new window from the widget editor
- [x] confirmed that the Stripe connection opens in the same window from the post editor, and that saving the post/etc continues to work

It does feel a bit odd that the form opens in a new window, because after connecting Stripe when the user returns to the original tab, they will still need to refresh the page before they can continue setting up the Payments button (the "Connect Stripe" button will still be there until refreshing). So they need to manually save, refresh, and then navigate back to their button. If they don't save before refreshing they'll get the same warning about navigating away.

It seems like the issue of navigating back to the specific widget being edited might be the issue? Regardless, I think it's an improvement to have the warning about losing unsaved changes only when the user refreshes.

**Review by stacimc — APPROVED** [review]
Sorry for the long time testing! LGTM 👍 Looks like the tests might just need to be re-run.

**Review by jeherve — APPROVED** [review]
Looks good to me. I think we may end up having to extract this into a new utility if this becomes used by other blocks maybe, but we'll cross that bridge when we get to it :)

**Comment by apeatling** [comment]
Thanks for merging @sdixon194 !

**Comment by jeherve** [comment]
r241943-wpcom

**Pull request merged**